examples/12_score.yaml

```diff
@@ -0,0 +1,69 @@
+# Using a trained model to add hypothesis score for an nbest list
+# First, exp1-model trains a model which is saved at examples/output/exp1-model.mod
+# Then, exp2-score loads the exp1-model, and use it to score an nbest list
+# The nbest list example used here is located at examples/data/head.nbest.en
+# exp2-score outputs a new nbest list with hypothesis score.
+# The output file will be in examples/output/exp2-score.test_hyp
+
+exp1-model: !Experiment
+  exp_global: !ExpGlobal
+    model_file: examples/output/{EXP}.mod
+    log_file: examples/output/{EXP}.log
+    default_layer_dim: 64
+    dropout: 0.5
+    weight_noise: 0.1
+  model: !DefaultTranslator
+    src_reader: !PlainTextReader
+      vocab: !Vocab {vocab_file: examples/data/head.ja.vocab}
+    trg_reader: !PlainTextReader
+      vocab: !Vocab {vocab_file: examples/data/head.en.vocab}
+    src_embedder: !SimpleWordEmbedder
+      emb_dim: 64
+    encoder: !BiLSTMSeqTransducer
+      layers: 2
+      input_dim: 64
+    attender: !MlpAttender
+      state_dim: 64
+      hidden_dim: 64
+      input_dim: 64
+    trg_embedder: !SimpleWordEmbedder
+      emb_dim: 64
+    decoder: !MlpSoftmaxDecoder
+      layers: 1
+      mlp_hidden_dim: 64
+      input_feeding: True
+      bridge: !CopyBridge {}
+    inference: !SimpleInference {}
+  train: !SimpleTrainingRegimen
+    run_for_epochs: 2
+    src_file: examples/data/head.ja
+    trg_file: examples/data/head.en
+    dev_tasks:
+    - !AccuracyEvalTask
+      eval_metrics: bleu
+      src_file: examples/data/head.ja
+      ref_file: examples/data/head.en
+      hyp_file: examples/output/{EXP}.dev_hyp
+  evaluate:
+    - !AccuracyEvalTask
+      eval_metrics: bleu
+      src_file: examples/data/head.ja
+      ref_file: examples/data/head.en
+      hyp_file: examples/output/{EXP}.test_hyp
+
+exp2-score: !Experiment
+  load: examples/output/exp1-model.mod
+  overwrite:
+  - path: exp_global.eval_only
+    val: True
+  - path: model.inference.mode
+    val: score
+  - path: model.inference.ref_file
```

Comment:
Maybe `nbest_file` would be a better name for this?

Reply:
Once this is fixed I'm happy to merge this! I think the failing example of the maximum recursion depth can be fixed in another commit.

Reply:
Hi,

I just merged my changes with the latest master commits. Should we merge after the loading error is fixed?